src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateCopy = {
+    ...state,
+  };
+  const previousVersions = [];
+
+  for (let i = 0; i < actions.length; i++) {
```

Comment:
consider to use array methods.
in this case forEach will be good
or `for of` loop